### PR TITLE
add keychain/1.2.0

### PR DESCRIPTION
**Pull request by hrantzsch**
Specify library name and version:  **keychain/1.2.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by hrantzsch** [comment]
Running the test package fails on Linux because libsecret needs the gnome-keyring-daemon (which I assume is only running in a full desktop session). Keychain CI kickstarts it for tests to run (see link in 084c52c), but for the test package I just remove the actual invocation.

File: recipes/keychain/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.includedirs = ['include']
+        self.cpp_info.libdirs     = ['lib']
```
Comment (intelligide) [inline recipes/keychain/all/conanfile.py:44]:
```suggestion
```

File: recipes/keychain/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+from conans import ConanFile, CMake, tools
+import os
+
+class KeychainConan(ConanFile):
+    name = "keychain"
+    homepage = "https://github.com/hrantzsch/keychain"
+    description = "A cross-platform wrapper for the operating system's credential storage"
+    topics = ("conan", "keychain", "security", "keychain", "password", "cpp11")
```
Comment (prince-chrismc) [inline recipes/keychain/all/conanfile.py:8]:
```suggestion
    topics = ("conan", "keychain", "security", "credentials", "password", "cpp11")
```

File: recipes/keychain/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+
+        if self.settings.os == 'Macos':
+            self.cpp_info.frameworks = ['Security', 'CoreFoundation']
+        if self.settings.os == "Windows":
+            self.cpp_info.system_libs = ['Crypt32']
```
Comment (prince-chrismc) [inline recipes/keychain/all/conanfile.py:52]:
```suggestion
            self.cpp_info.system_libs = ['crypt32']
```

File: recipes/keychain/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
+set_target_properties(${PROJECT_NAME} PROPERTIES CXX_STANDARD 11 CXX_STANDARD_REQUIRED ON)
```
Comment (prince-chrismc) [inline recipes/keychain/all/test_package/CMakeLists.txt:9]: if this is required you are missing 

https://github.com/conan-io/conan-center-index/blob/8d389aa0c380f1ea7d0cb9324768ad6051c04874/recipes/elfio/all/conanfile.py#L19-L21

**Review by prince-chrismc — CHANGES_REQUESTED** [review]
Super close just small details!

File: recipes/keychain/all/test_package/CMakeLists.txt (hunk 4 above)
Comment (hrantzsch) [reply recipes/keychain/all/test_package/CMakeLists.txt:9]: fixed, thanks!

**Review by uilianries — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**